### PR TITLE
chore(refactor): refactored everything for newer typescript and easie…

### PR DESCRIPTION
**Pull request by sjelin**
…r writing of simple commands

* Use `@types/*` to get types instead of `typings`
* Upgrade to typescript 2
* Update gulpfile for the above
* Use Column Limit 100 with clang
* Use DefinitelyTyped typings for WebDriver
* Generally tighten up types
* Update some URLs in comments to reference specific versions of the selenium codebase
* Prefer `_` to camel case in file names
* Create `SimpleCommand` class to reduce boilerplate code
* Move definitions for simple commands into a separate file to reduce clutter in index.js

### CODE REVIEW

File: gulpfile.js [hunk 1]
```diff
@@ -30,24 +28,26 @@ gulp.task('copy', function() {
...
 });
 
-gulp.task('clang', function() {
-  return gulp.src(['src/**/*.ts'])
-      .pipe(gulpFormat.checkFormat('file', clangFormat))
-      .on('warning', function(e) {
-    console.log(e);
-  });
+gulp.task('format:enforce', () => {
+  const format = require('gulp-clang-format');
```
Comment (juliemr) [inline gulpfile.js]: Why pull these `require`s down here? Seems like this causes duplication.

File: lib/simple_command.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import {promise as Promise} from 'selenium-webdriver';
+
+import {Extender} from './extender';
+
+export class SimpleCommand<T> {
+  name: string;
+  params: string[];
+  method: string;
+  path: string;
+  constructor(name: string, params: string[], method: string, path: string) {
```
Comment (juliemr) [inline lib/simple_command.ts]: You can shortcut this with typescript:

```ts
export class SimpleCommand<t> {
  constructor(public name: string, public params: string[], public method: string, public path: string) {}
```

File: gulpfile.js (hunk 1 above)
Comment (sjelin) [inline gulpfile.js]: I copied this from [the `webdriver-manager` gulp file](https://github.com/angular/webdriver-manager/blob/master/gulpfile.js#L34), which I think @cnishina copied from someone else.  I'd be happy to move it.

**Comment by sjelin** [comment]
@juliemr comments addressed

**Comment by juliemr** [comment]
LGTM!

**Pull request merged**